lib/linguist/languages.yml

```diff
@@ -2847,6 +2847,7 @@ JavaScript:
   - ".gs"
   - ".jake"
   - ".javascript"
+  - ".js.frag"
```

Comment:
Whoops, looks like you left this over from your original commits. This isn't needed. It'll be caught by the `.frag` above and your new heuristic.

Reply:
No, I wanted to make it so that `.js.frag` enforces the usage of JS syntax highlighting, as I imagine old projects would do that (because if you treat the `.frag` extension as meaning "fragment of a file" , then naming your file `start.frag` in a Javascript project is like naming a file `program.txt` and expecting the syntax highlighting to work). Also, I wrote a heuristic only for a start of an IIFE (I imagine that the vast majority of usages of `.frag` for JS are for module wrappers as I have explained in the first post) because, in my opinion, JS should be the fallback choice here. Although GLSL fragment shaders have a few much more recognizable keywords, and a heuristic for them already exists, so it might be more reasonable to use that. What do you think?

Reply:
The language detection, by whatever means (ext, heuristic etc) determines the syntax highlighting. We can add this extension but then you need to fulfill all the requirements for adding a new extension.